peer.go

```diff
@@ -878,6 +891,14 @@ func newDiscMsgStream(p *peer) *msgStream {
 //
 // NOTE: This method MUST be run as a goroutine.
 func (p *peer) readHandler() {
+	chanMsgStreams := make(map[lnwire.ChannelID]*msgStream)
+
+	defer p.wg.Done()
+	defer func() {
+		for _, chanStream := range chanMsgStreams {
+			chanStream.Stop()
```

Comment:
Why not defer right after each is started?

Reply:
Fixed.